src/visitable.cpp

```diff
@@ -774,7 +782,8 @@ static int amount_of_internal( const T& self, const itype_id& id, bool pseudo, i
 {
     int qty = 0;
     self.visit_items( [&qty, &id, &pseudo, &limit] ( const item *e ) {
-        qty += ( e->typeId() == id && e->allow_crafting_component() && ( pseudo || !e->has_flag( "PSEUDO" ) ) );
+        bool is_component = e->typeId() == id && e->allow_crafting_component() && ( pseudo || !e->has_flag( "PSEUDO" ) );
+        qty = accumulate_with_limit( qty, is_component, limit );
```

Comment:
The original line is questionable but this is even more opaque.

You parse that as:

`qty = accumulate_with_limit( int, bool, int )`

But it is actually of course

`qty = accumulate_with_limit( int, int, int )`

The more I think about it we should be using operators otherwise we are going to end up with lots of nasty casts like this or even worse functions like `add_one`, `minus_one`.